bigtable/client/internal/table_admin.h

```diff
@@ -117,6 +120,20 @@ class TableAdmin {
                         bigtable::ConsistencyToken const& consistency_token,
                         grpc::Status& status);
 
+  std::future<bool> WaitForConsistencyCheck(
```

Comment:
Having a `noex` version of these functions does not work.  `std::future<>` requires `std::exception_ptr`, the only way to signal errors through them is to raise an exception.  I think we should not implement the `noex` versions of the future-returning functions.

Reply:
I will remove  the noex version and implement whole functionality in bigtable/client/table_admin.h

Reply:
I have implemented it as follows.
1. internal/table_admin.h has a friend function WaitForConsistencyCheck, it has implementation as before with additional parameter as TableAdmin object.
2. All other functions from internal/table_admin.h are removed.
3. Asynchronous test cases from internal/table_admin_test.cc are removed.
4. client/table_admin.h has functions as it is.
5. client/table_admin.h has a function WaitForConsistencyCheckImpl which calls friend function from internal/table_admin.h and passes impl_ to the friend function as a object of noex::TableAdmin.

Is it ok? or any other suggestions.

Reply:
> internal/table_admin.h has a friend function WaitForConsistencyCheck, it has implementation
> as before with additional parameter as TableAdmin object.

That sounds odd to me.  Why do we want any code in `internal/table_admin.h` if we are only going to use it in `client/table_admin.{h,cc}`?

> All other functions from internal/table_admin.h are removed.
> Asynchronous test cases from internal/table_admin_test.cc are removed.
> `client/table_admin.h` has functions as it is.

Makes sense.

> client/table_admin.h has a function WaitForConsistencyCheckImpl which calls friend function
> from internal/table_admin.h and passes impl_ to the friend function as a object of
> noex::TableAdmin.

Again, that sounds odd.  Feel free to share the code and I can comment further, but I think you want to move that code from `noex` to the `WaitForConsistencyCheckImpl()` function

Reply:
The issue to move whole code from noex::internal/table_admin to client/table_admin is that, all policies and major function implementations is in the noex::internal::TableAdmin class, and client/TableAdmin only contains object of noex::TableAdmin class.
If I need to move the functionality then I need major changes in client/TableAdmin class.

Reply:
Ahhh, good point.  Then you need to implement `WaitForConsistencyCheckImpl` in the `bigtable::noex::TableAdmin` class, as you said earlier.